app/jobs/delete_pii_from_old_claims_job.rb

```diff
@@ -0,0 +1,20 @@
+# Runs weekly at 00:30 on a Sunday morning, and requests that eligible claims
+# (rejected or paid, and more than 60 days since their last update) have their
+# PII scrubbed.
+class DeletePiiFromOldClaimsJob < CronJob
+  self.cron_expression = "30 0 * * 0"
+
+  MINIMUM_TIME_SINCE_LAST_UPDATE = 2.months.ago
+
+  def perform
+    Rails.logger.info "Deleting PII from #{claims_eligible_to_delete.count} claims"
+    Claim::Cleaner.new(claims_eligible_to_delete).delete_pii
+  end
+
+  private
+
+  def claims_eligible_to_delete
```

Comment:
I'd prefer to see this in one query, that way it returns an `ActiveRecord_Relation`, which we can then chain additional queries onto. I've had a bit of a play and this seems to do the trick:

```ruby
Claim.left_outer_joins(payment: [:payroll_run])
  .joins(:decision)
  .where(
    "(decisions.result = 1 AND claims.updated_at < :minumum_time) OR scheduled_payment_date < :minumum_time",
    minumum_time: MINIMUM_TIME_SINCE_LAST_UPDATE
  )
```

Reply:
It might be worth giving a more descriptive name for the method too - something like `old_rejected_or_paid_claims`?

Reply:
Have a look if you can make use of the [`or` method](https://blog.bigbinary.com/2016/05/30/rails-5-adds-or-support-in-active-record.html) to chain this such that we don't need to have a hardcode the enum mapping ("decisions.result = 1") and instead pass in the `Claim.rejected` scope.

Reply:
We'll also want to exclude claims that have already had their PII removed from the relation right?

Reply:
I tried using `or`, and this resulted in an `ArgumentError (Relation passed to #or must be structurally compatible. Incompatible values: [:left_outer_joins])`